src/query/stream/stream_consumer_factory.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+#pragma once
+
+#include "query/interpreter_context.hpp"
+#include "query/stream/istream_consumer_factory.hpp"
+
+namespace memgraph::query::stream {
+struct StreamConsumerFactory : IStreamConsumerFactory {
```

Comment:
Why introduce virtual interface if we have just one impl? I don't think this will change soon in the future?

Reply:
Its the whole point of this, to decouple stream evaluation from query. The plan is to untangle, dbms/database/query/streams/triggers so that system transaction can exist independently.